src/common/config.h

```diff
@@ -343,6 +343,9 @@ struct md_config_t {
   unsigned get_osd_pool_default_min_size(const ConfigValues& values,
                                          uint8_t size) const {
     uint8_t min_size = get_val<uint64_t>(values, "osd_pool_default_min_size");
+    // Special case because 3 - 3/2 == 1
+    if (min_size == 0 && size == 3)
+      return 2;
     return min_size ? std::min(min_size, size) : (size - size / 2);
```

Comment:
3/2 should be 1 for integer arithmetic, so this should result in min_size 2 already

Reply:
Yes, I agree this is what I remember when it is set to 0 as default it uses formula `size - size / 2`. The same is mentioned in the option.cc.

~~~
 Option("osd_pool_default_min_size", Option::TYPE_UINT, Option::LEVEL_ADVANCED)
    .set_default(0)
    .set_min_max(0, 255)
    .set_flag(Option::FLAG_RUNTIME)
    .set_description("the minimal number of copies allowed to write to a degraded pool for new replicated pools")
    .set_long_description("0 means no specific default; ceph will use size-size/2")
    .add_see_also("osd_pool_default_size")
    .add_service("mon"),
~~~

That has been the case in deployments also if we set size as 3 min_size comes as 2 until specifically not specified value as 1 for this option `osd_pool_default_min_size`.

Reply:
@dzafman you've probably only seen this on a vstart cluster because we explicitly set `osd_pool_default_min_size = 1` there, which overrides all this logic

Reply:
@neha-ojha Do you mean that the testing sets `osd_pool_default_min_size = 1`.  If so, this isn't a bug found in testing.  It is an issue in production.  If you are setting a pool to size 3 and haven't set a default min_size which arguably should always be 2 or greater for replicated, then we should override the size - size /2 case.

Reply:
In production by default osd_pool_detault_min_size == 0

```
Option("osd_pool_default_min_size", Option::TYPE_UINT, Option::LEVEL_ADVANCED)
    .set_default(0)
```

Reply:
@dzafman See https://tracker.ceph.com/issues/46325#note-3